src/forms/Checkbox.jsx

```diff
@@ -41,103 +32,51 @@ class Checkbox extends React.PureComponent {
-		this.setState({ focused: true });
-	}
-
 	getChecked() {
-		if (this.props.controlled) return this.state.checked;
+		if (this.props.controlled) {
+			return this.state.checked;
+		}
+
 		return this.props.checked;
 	}
 
 	render() {
 		const {
 			checked, // eslint-disable-line no-unused-vars
+			controlled, // eslint-disable-line no-unused-vars
 			children,
-			className,
-			labelClassName,
 			id,
 			label,
-			name,
-			value,
 			disabled,
-			onBlur, // eslint-disable-line no-unused-vars
-			onFocus, // eslint-disable-line no-unused-vars
-			onChange, // eslint-disable-line no-unused-vars
-			controlled, // eslint-disable-line no-unused-vars
-			iconShape,
+			name, // TODO: should v2 Checkbox support 'name'?
+			value, // TODO: Should v2 Checkbox support 'value'?
+			// iconShape, // TODO: should v2 Checkbox support 'iconShape'?
 			...other
 		} = this.props;
-		const stateChecked = this.getChecked();
-
-		const classNames = cx('minTouchHeight', className);
-
-		const labelClassNames = cx(
-			'toggleLabel label--minor display--block',
-			labelClassName
-		);
-
-		const fauxCheckboxClassNames = cx(
-			'display--block align--center',
-			FAUX_TOGGLE_CLASS,
-			`${FAUX_TOGGLE_CLASS}--checkbox`,
-			{
-				checked: stateChecked,
-				disabled,
-				[FOCUSED_CHECKBOX_CLASS]: this.state.focused,
-				'checked--disabled': stateChecked && disabled,
-				'checked--focused': this.state.focused && disabled,
-			}
-		);
 
 		const elId = id || `${name}-${value}`;
+		const stateChecked = this.getChecked();
 
-		return (
-			<label className={labelClassNames} htmlFor={elId}>
-				<Flex align="center" className={classNames} {...other}>
-					<FlexItem shrink>
-						<input
-							type="checkbox"
-							name={name}
-							value={value}
-							checked={stateChecked}
-							disabled={disabled}
-							className="checkbox visibility--a11yHide"
-							id={elId}
-							onBlur={this.onBlur}
-							onFocus={this.onFocus}
-							onChange={this.onChange}
-						/>
-						<span
-							ref={el => (this.fauxCheckboxEl = el)}
-							className={fauxCheckboxClassNames}
-						>
-							{stateChecked && (
-								<Icon
-									className="display--flex flex--center checkbox-indicator"
-									shape={iconShape || 'check'}
-									size="xs"
-								/>
-							)}
-						</span>
-					</FlexItem>
-					<FlexItem className="toggleLabel-container">
-						<span
-							className={cx({
-								['text--hint']: disabled,
-								['text--bold']: stateChecked,
-							})}
-						>
-							{label}
-						</span>
-						{children}
-					</FlexItem>
-				</Flex>
-			</label>
+		return children ? (
+			<SwarmCheckbox
+				checked={stateChecked}
+				label={label}
+				id={elId}
+				disabled={disabled}
+				onChange={this.onChange}
+				{...other}
+			>
+				{children}
+			</SwarmCheckbox>
+		) : (
+			<SwarmCheckbox
+				checked={stateChecked}
+				label={label}
+				id={elId}
+				disabled={disabled}
+				onChange={this.onChange}
+				{...other}
+			/>
```

Comment:
What about DRYing this up a bit so that if someone needs to change a prop they do it in once place and don't forget to do it in both (like the `checked` prop bug we fixed when we paired)

Something like this instead:

```javascript
		const swarmCheckboxProps = {
				label,
				disabled,
				checked: stateChecked,
				id: elId,
				onChange: this.onChange
		}

		return children ? (
			<SwarmCheckbox
				{...swarmCheckboxProps}
				{...other}
			>
				{children}
			</SwarmCheckbox>
		) : (
			<SwarmCheckbox
				{...swarmCheckboxProps}
				{...other}
			/>
```

Reply:
this looks odd - can't we just pass the `children` into `SwarmCheckbox` regardless of whether it's falsey? In that case, we don't actually need the value of `children` and can remove it from the destructuring assignment and just let it pass through along with `other`, which would make this a lot simpler

Reply:
Agreed, this could be simpler. We could omit a few of the props from the destructuring in render and allow them to pass through in other winding up with something like 
```jsx
return (
			<SwarmCheckbox
				checked={this.getChecked()}
				id={elId}
				disabled={disabled}
				onChange={this.onChange}
				{...other}
			/>
		);
```

Reply:
Does passing along `children` as part of `other` actually work? I tried to look this up but couldn't find anything in React docs or otherwise about it.

Reply:
Yes, it's treated the same as other props in react